### PR TITLE
Customer Intelligence Pre-Populution

### PR DESCRIPTION
**Pull request by 00eadeyemi**
### Added

 - Customer Intelligence information now pre-populations assisted journey forms when available (#164102346)

[Delievers[#164102346](https://www.pivotaltracker.com/story/show/164102346)]

### CODE REVIEW

File: app/Http/Controllers/InitialisationController.php [hunk 1]
```diff
@@ -643,4 +653,67 @@ private function checkIfProfilePageCanBeViewed(Application $application)
...
+        try {
+            if ($preApprovalApplicant = $this->getPreApprovalApplicant($request, $location)) {
+                return $preApprovalApplicant;
+            }
+
+            if ($leadScoreApplicant = $this->getLeadScoreApplicant($request, $location)) {
+                return $leadScoreApplicant;
+            }
+        } catch (\Exception $e) {
+            $this->logInfo('Unable to retrieve customer intelligence applicant for pre-propulation ' . $e->getMessage());
```
Comment (danieljjy) [inline app/Http/Controllers/InitialisationController.php]: Spelling

File: app/Basket/Helpers/FormHelper.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+class FormHelper
+{
+    /**
+     * @author GK
+     * @author EA
+     *
+     * Encapsulates php's built in range function to make the keys match the values
+     *
+     * @param float|int $low Low value.
+     * @param float|int $high High value.
```
Comment (KrekoG) [inline app/Basket/Helpers/FormHelper.php]: Just been told by SL that these are actually not right, as in it trips some automated systems. (Even tho the code was originally from me, so my bad)

It should be in the following format:
```
/**
     * Encapsulates php's built in range function to make the keys match the values
     *
     * @author GK
     * @author EA
     * @param float|int $low
     * @param float|int $high
```

So the @tags together and with no comment on the parameters, which is annoying.

File: public/js/profile.js [hunk 3]
```diff
@@ -145,16 +145,7 @@ $(document).ready(function() {
         }
     });
 
-    $('#dob_day').prepend( '<option value="">-- Day --</option>');
-    $('#dob_month').prepend( '<option value="">-- Month --</option>');
-    $('#dob_year').prepend( '<option value="">-- Year --</option>');
-    $('#dob_day :nth-child(1)').prop('selected', true);
-    $('#dob_month :nth-child(1)').prop('selected', true);
-    $('#dob_year :nth-child(1)').prop('selected', true);
```
Comment (KrekoG) [inline public/js/profile.js]: As much as I looove seeing this bit of code being thrown out of the window, have you searched the project, so we are sure it is not used anywhere else?

File: resources/views/initialise/assisted.blade.php [hunk 4]
```diff
@@ -58,20 +58,20 @@
                                 <div class="col-sm-8">
                                     <div class="input-group">
                                         <div class="input-group-addon"><i class="glyphicon glyphicon-envelope"></i></div>
-                                        {!! Form::email('email', null, ['class' => 'form-control col-xs-12', 'data-fv-notempty' => 'true', 'data-fv-notempty-message' => 'Please enter an email address', 'data-fv-emailaddress' => 'true', 'maxlength' => 255]) !!}
+                                        {!! Form::email('email', ($applicant->getEmail() != null ? $applicant->getEmail() : null), ['class' => 'form-control col-xs-12', 'data-fv-notempty' => 'true', 'data-fv-notempty-message' => 'Please enter an email address', 'data-fv-emailaddress' => 'true', 'maxlength' => 255]) !!}
```
Comment (KrekoG) [inline resources/views/initialise/assisted.blade.php]: If I'm following correctly, If the applicant's e-mail isn't null, we return the e-mail, if it is null, we return null. Could this not be simplified by just returning the e-mail regardless of it's value?

**Pull request merged**